src/dmd/frontend.d

```diff
@@ -54,3 +62,161 @@ void initDMD()
     builtin_init();
 }
 
+/**
+Add import paths to the `global.path`.
+Params:
+    path = range of imports to add
+*/
+void addImports(T)(T path)
+{
+    import dmd.globals : global;
+    import dmd.arraytypes : Strings;
+
+    Strings* res = new Strings();
+    foreach (p; path)
+    {
+        import std.string : toStringz;
+        Strings* a = new Strings();
+        a.push(p.toStringz);
+        res.append(a);
+    }
+    global.path = res;
+}
+
+/**
+Finds a dmd.conf and parses it for import paths.
+This depends on the `$DMD` environment variable.
+If it's set to `ldmd`, it will try to detect and parse an `ldc2.conf` instead.
+
+Returns:
+    A forward range of normalized import paths.
+*/
+auto findImportPaths()
+{
+    import std.file : exists, getcwd;
+    import std.string : fromStringz, toStringz;
+    import std.path : buildPath, buildNormalizedPath, dirName;
+    import std.process : env = environment, execute;
+    import dmd.dinifile : findConfFile;
+    import dmd.errors : fatal;
+    import std.algorithm, std.range, std.regex;
+    import std.stdio : File,  stderr;
+
+    auto dmdEnv = env.get("DMD", "dmd");
```

Comment:
I think this function needs to be split into several layers building on top of each other. One where it's possible to give a direct path to the config and another which tries to infer it.